File: src/sizeof_types.c
#include <stdio.h>

int main() {
    // Affichage des tailles pour les types de base
    printf("Taille de char : %zu octets\n", sizeof(char));
    printf("Taille de signed char : %zu octets\n", sizeof(signed char));
    printf("Taille de unsigned char : %zu octets\n", sizeof(unsigned char));

    printf("Taille de short : %zu octets\n", sizeof(short));
    printf("Taille de signed short : %zu octets\n", sizeof(signed short));
    printf("Taille de unsigned short : %zu octets\n", sizeof(unsigned short));

    printf("Taille de int : %zu octets\n", sizeof(int));
    printf("Taille de signed int : %zu octets\n", sizeof(signed int));
    printf("Taille de unsigned int : %zu octets\n", sizeof(unsigned int));

    printf("Taille de long int : %zu octets\n", sizeof(long int));
    printf("Taille de signed long int : %zu octets\n", sizeof(signed long int));
    printf("Taille de unsigned long int : %zu octets\n", sizeof(unsigned long int));

    printf("Taille de long long int : %zu octets\n", sizeof(long long int));

    printf("Taille de float : %zu octets\n", sizeof(float));
    printf("Taille de double : %zu octets\n", sizeof(double));
    printf("Taille de long double : %zu octets\n", sizeof(long double));

    return 0;
}

File: src/sizeof_types.py
import sys

# Définition des types de base
int_value = 1  # Un entier
float_value = 1.9  # Un flottant
bool_value = True  # Une valeur booléenne
str_value = "Hell"  # Une chaîne de caractères
list_value = [1, 2, 3, 4, 5]  # Une liste
tuple_value = (1, 2, 3, 4, 5)  # Un tuple
dict_value = {'a': 0, 'b': 5, 'c': 3}  # Un dictionnaire

# Affichage de la taille de chaque objet en mémoire
print(f"Taille d'un int : {sys.getsizeof(int_value)} octets")
print(f"Taille d'un float : {sys.getsizeof(float_value)} octets")
print(f"Taille d'un bool : {sys.getsizeof(bool_value)} octets")
print(f"Taille d'une str : {sys.getsizeof(str_value)} octets")
print(f"Taille d'une list : {sys.getsizeof(list_value)} octets")
print(f"Taille d'un tuple : {sys.getsizeof(tuple_value)} octets")
print(f"Taille d'un dict : {sys.getsizeof(dict_value)} octets")
